integration_tests/commands/resp/pfcountwatch_test.go

```diff
@@ -0,0 +1,194 @@
+
+	defer func() {
+		if err := publisher.Close(); err != nil {
+			t.Errorf("Error closing publisher connection: %v", err)
+		}
+		for _, sub := range subscribers {
+			time.Sleep(100 * time.Millisecond)
+			if err := sub.Close(); err != nil {
+				t.Errorf("Error closing subscriber connection: %v", err)
+			}
+		}
+	}()
+
+	res := FireCommand(publisher, fmt.Sprintf("PFADD %s %s", pfcountWatchKey, "randomvalue"))
+	assert.Equal(t, int64(1), res)
+
+	respParsers := make([]*clientio.RESPParser, len(subscribers))
+	for i, subscriber := range subscribers {
+		rp := fireCommandAndGetRESPParser(subscriber, fmt.Sprintf(pfcountWatchQuery, pfcountWatchKey))
+		assert.Assert(t, rp != nil)
+		respParsers[i] = rp
+
+		v, err := rp.DecodeOne()
+		assert.NilError(t, err)
+		castedValue, ok := v.([]interface{})
+		if !ok {
+			t.Errorf("Type assertion to []interface{} failed for value: %v", v)
+		}
+		assert.Equal(t, 3, len(castedValue))
+	}
+
+	// Add elements with PFADD
+	for _, tc := range pfcountWatchTestCases {
+		res := FireCommand(publisher, fmt.Sprintf("PFADD %s %s", tc.key, tc.val))
+		assert.Equal(t, int64(1), res)
+		for _, rp := range respParsers {
+			v, err := rp.DecodeOne()
+			assert.NilError(t, err)
+			castedValue, ok := v.([]interface{})
+			if !ok {
+				t.Errorf("Type assertion to []interface{} failed for value: %v", v)
+			}
+			assert.Equal(t, 3, len(castedValue))
+			assert.Equal(t, "PFCOUNT", castedValue[0])
+			assert.Equal(t, "1580567186", castedValue[1])
+			assert.DeepEqual(t, tc.result, castedValue[2])
+		}
+	}
+
+	// Test for PFMERGE Case
+	for _, tc := range pfcountWatchhWithPFMergeTestCases {
+		FireCommand(publisher, fmt.Sprintf("DEL %s %s", tc.dest_key_1, tc.dest_key_2))
+		FireCommand(publisher, fmt.Sprintf("PFADD %s %s", tc.dest_key_1, tc.dest_value_1))
+		FireCommand(publisher, fmt.Sprintf("PFADD %s %s", tc.dest_key_2, tc.dest_value_2))
+		FireCommand(publisher, fmt.Sprintf("PFMERGE %s %s %s", pfcountWatchKey, tc.dest_key_1, tc.dest_key_2))
+		assert.Equal(t, int64(1), res)
+		for _, rp := range respParsers {
+			v, err := rp.DecodeOne()
+			assert.NilError(t, err)
+			castedValue, ok := v.([]interface{})
+			if !ok {
+				t.Errorf("Type assertion to []interface{} failed for value: %v", v)
+			}
+			assert.Equal(t, 3, len(castedValue))
+			assert.Equal(t, "PFCOUNT", castedValue[0])
+			assert.Equal(t, "1580567186", castedValue[1])
+			assert.DeepEqual(t, tc.result, castedValue[2])
+		}
+	}
+}
+
+type pfcountWatchSDKTestCase struct {
+	key    string
+	val    any
+	result int64
+}
+
+type pfcountWatchSDKWithPFMergeTestCase struct {
+	dest_key_1    string
+	dest_value_1  []string
+	dest_key_2 	  string
+	dest_value_2  []string
+	result 		  int64
+}
+
+var PFCountWatchSDKTestCases = []pfcountWatchSDKTestCase{
+	{pfcountWatchKey, "value1", int64(1)},
+	{pfcountWatchKey, "value2", int64(2)},
+	{pfcountWatchKey, "value3", int64(3)},
+	{pfcountWatchKey, "value4", int64(4)},
+}
+
```

Comment:
Ideally we should have constants defined separately.
```suggestion
pfCountWatchKey  = "pfcount_watch_key" // assuming this was defined elsewhere
defaultTimeout   = 5 * time.Second
watchFingerprint = "1580567186"
```

Reply:
Have defined constants differently for SDK test.